src/views/DatePickerBaseView.ts

```diff
@@ -1,34 +1,69 @@
 /*
  These views only present dates, not navigation
  */
-
-import ContinuousScrollHandler from "./virtualDom/ContinuousScrollHandler";
+import AtomicDateObject from "../models/AtomicDateObject";
 import DatePickerFactory from "../models/datePickerFactory/DatePickerFactory";
-import VirtualDom from "./virtualDom/VirtualDom";
+import VirtualDom from "./VirtualDom";
 
 export default abstract class DatePickerBaseView {
+  continuousScroll: boolean;
+
   protected frameElement: HTMLElement;
+  protected frameElementClassName: string;
   protected model: DatePickerFactory;
   protected virtualDom: VirtualDom;
 
-  protected initFrameView(continuousScroll = true, looping = false): HTMLElement {
+  constructor(model: DatePickerFactory, continuousScroll = true) {
+    this.continuousScroll = continuousScroll;
+    this.model = model;
+  }
+
+  protected initFrameView(continuousScroll = true): HTMLElement {
     this.frameElement = document.createElement("div");
+    console.log("GRAND SUPER");
     if (continuousScroll) {
-      // init ContinuousScrollHandler
-      this.virtualDom = new VirtualDom(new ContinuousScrollHandler(this.model))
-      this.frameElement.appendChild(this.virtualDom.frameElement)
+      this.virtualDom = new VirtualDom();
+      this.frameElement.appendChild(this.virtualDom.frameElement);
+      console.log("---- VIRTUAL DOM");
+    } else {
+      console.log("no virtual dom");
     }
 
     this.frameElement.className += " date-picker-view";
     return this.frameElement;
   }
 
-  abstract append(parentElement: HTMLElement): void;
+  append(parentElement: HTMLElement): void {
+    this.frameElement = this.initFrameView(this.continuousScroll);
+    const classNames = this.frameElement.className.split("");
+    classNames.push(this.frameElementClassName);
+    this.frameElement.className = classNames.join(" ");
```

Comment:
Hello! I am missing something here. When I change the date picker control to a type that uses this block, I'm seeing `this.frameElement.className` end up as `d a t e - p i c k e r - v i e w date-picker-list`. What's the reasoning behind splitting `this.frameElement.className`into individual letters?

Reply:
Hi! I'm not able to replicate this. Can you tell me what you did?

Reply:
Changed `dateType` to `1` in index.ts, then sent the output of `this.frameElement.className` to console after line 40.

Reply:
Awesome catch!!! Thank you!